Xilinx platform mangement related changes

sivadur opened the pull request:
This series contains xilinx platform/power management related changes which are fixes/ minor feature additions.

antonio-nino-diaz-arm commented on plat/xilinx/zynqmp/aarch64/zynqmp_common.c:
```diff
@@ -59,6 +61,7 @@ static const struct {
 	unsigned int id;
 	unsigned int ver;
 	char *name;
+	bool evexists;
```
nit: It's a bit weird to use a `bool` if you are going to assign to it integer values (1 or 0).

sivadur replied:
Thanks for your comment, will use true or false instead of 1 or 0.

sivadur commented:
@antonio-nino-diaz-arm 

Updated with your comment .
Also, i added a new patch to the series for handling wdt reset incase of wdt timeout in the system.

antonio-nino-diaz-arm commented on plat/xilinx/zynqmp/pm_service/pm_svc_main.c:
```diff
@@ -32,6 +45,144 @@ static struct {
+				0xffffffff);
+	}
+
+	dsb();
```
This `dsb` isn't needed, the store release inside `spin_unlock` already enforces the previous writes to be seen before the write that unlocks the spinlock.

Check "K10.2.1 Message passing" in the Arm ARM (version B.b). The example "Resolving weakly-ordered message passing by using Acquire and Release" is a very simple one and explains this.

antonio-nino-diaz-arm commented:
This PR needs a rebase.

sivadur commented:
@antonio-nino-diaz-arm  Sure, I will fix review comments and will rebase while doing it.
